GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -158,6 +158,58 @@ public virtual void RevokeCredential(string repoUrl)
                 null);
         }
 
+        /// <summary>
+        /// Input for certificate crednetials looks like
+        /// <code> protocol=cert
+        /// path=[http.sslCert value]
+        /// username =</code>
+        /// </summary>
+        public virtual bool TryGetCertificatePassword(
+            ITracer tracer,
+            string certificatePath,
+            out string password,
+            out string errorMessage)
+        {
+            password = null;
+            errorMessage = null;
+
+            using (ITracer activity = tracer.StartActivity("TryGetCertificatePassword", EventLevel.Informational))
+            {
+                Result gitCredentialOutput = this.InvokeGitAgainstDotGitFolder(

```

Comment:
Have you tested this on Windows?

Reply:
No, sadly I do not have access to a windows machine. Do you think something in particular may behave differently here?

Reply:
Prompting for a password from potentially a background process is just a flow we don't have good automation around, which is why I was asking.

Reply:
IMO, this doesn't differ from `TryGetCredentials`, a method you already have. It invokes the same git credential helper, with the same potential of being invoked from a background process. So the way I see it, I'm not introducing anything new here. And yes, when credentials change, while VFS4G is running, and when getting new credentials requires user interaction in terminal (be it for certificate or for http), bad things will happen, but that's something, that is already in the code.

Reply:
@wilbaker do you agree with this? Or do you think that my changes introduce some additional risks?